GPU/Vulkan/TextureCacheVulkan.cpp

```diff
@@ -400,27 +399,43 @@ void TextureCacheVulkan::ApplyTextureFramebuffer(TexCacheEntry *entry, VirtualFr
 			const float top = v1 * invHalfHeight - 1.0f;
 			const float bottom = v2 * invHalfHeight - 1.0f;
 			// Points are: BL, BR, TR, TL.
-			pos[0] = Pos(left, bottom, -1.0f);
-			pos[1] = Pos(right, bottom, -1.0f);
-			pos[2] = Pos(right, top, -1.0f);
-			pos[3] = Pos(left, top, -1.0f);
+			verts[0].x = left;
+			verts[0].y = bottom;
+			verts[1].x = right;
+			verts[1].y = bottom;
+			verts[2].x = left;
+			verts[2].y = top;
+			verts[3].x = right;
+			verts[3].y = top;
 
 			// And also the UVs, same order.
 			const float uvleft = u1 * invWidth;
 			const float uvright = u2 * invWidth;
 			const float uvtop = v1 * invHeight;
 			const float uvbottom = v2 * invHeight;
-			uv[0] = UV(uvleft, uvbottom);
-			uv[1] = UV(uvright, uvbottom);
-			uv[2] = UV(uvright, uvtop);
-			uv[3] = UV(uvleft, uvtop);
+			verts[0].u = uvleft;
+			verts[0].v = uvbottom;
+			verts[1].u = uvright;
+			verts[1].v = uvbottom;
+			verts[2].u = uvleft;
+			verts[2].v = uvtop;
+			verts[3].u = uvright;
+			verts[3].v = uvtop;
 		}
 
-		shaderManagerVulkan_->DirtyLastShader();
+		VkBuffer pushed;
+		uint32_t offset = push_->PushAligned(verts, sizeof(verts), 4, &pushed);
+
+		draw_->BindFramebufferAsTexture(framebuffer->fbo, 0, Draw::FB_COLOR_BIT, 0);
+		VkImageView fbo = (VkImageView)draw_->GetNativeObject(Draw::NativeObject::BOUND_TEXTURE0_IMAGEVIEW);
 
-		//depalFBO->EndPass(cmd);
-		//depalFBO->TransitionToTexture(cmd);
-		//imageView = depalFBO->GetColorImageView();
+		VkDescriptorSet descSet = vulkan2D_->GetDescriptorSet(fbo, samplerNearest_, clutTexture->GetImageView(), samplerNearest_);
+		VulkanRenderManager *renderManager = (VulkanRenderManager *)draw_->GetNativeObject(Draw::NativeObject::RENDER_MANAGER);
+		renderManager->BindPipeline(depalShader->pipeline);
+		renderManager->SetScissor(VkRect2D{ {0, 0}, { framebuffer->renderWidth, framebuffer->renderHeight} });
```

Comment:
Do we dirty this state afterward? I may have missed it.

-[Unknown]

Reply:
Nope. Fixing, thanks.